forge-gui-desktop/src/main/java/forge/view/Main.java

```diff
@@ -77,23 +77,19 @@ public static void main(final String[] args) {
             case "server":
                 System.out.println("Dedicated server mode.\nNot implemented.");
                 break;
-            
+
             default:
                 System.out.println("Unknown mode.\nKnown mode is 'sim', 'parse' ");
                 break;
         }
-        
-        System.exit(0);
-    }
 
-    @SuppressWarnings("deprecation")
-	@Override
-    protected void finalize() throws Throwable {
         try {
             ExceptionHandler.unregisterErrorHandling();
-        } finally {
-            super.finalize();
+        } catch (java.io.IOException e) {
```

Comment:
This here could potentially be a logic change in case of no command line arguments (since we return then and don't have this extra function)
It's nice if we can get rid of something deprecated, but the call to `unregisterErrorHandling()` seems to release log file access among other things
Though unfortunately I don't know this part here well enough to judge if this was even supposed to work like that during normal execution, but I'll try to investigate 🤔

Reply:
All right, I'm currently assuming you stumbled upon something that hasn't been fully working for a while (at least on my setup)
Now it's a bit hard to tell how bad of a resource leak this could actually cause, but ideally the call can be duplicated during shutdown logic of the GUI to be safe
So while I don't really want to stall this - it would probably make sense if we try to address this here too
(as long as the task won't be too complex)

Reply:
Would you lay out steps for us to follow?
P.S.: In running main with CLI arguments `sim -d Virtue_And_Valor.dck Virtue_And_Valor.dck`, two AI's play a game. What CLI arguments do I use to launch the GUI?

Reply:
Uhh, that logic is just above the part you've been editing:
https://github.com/Card-Forge/forge/blob/4c0599bb52ec719e5b06cbde653cd849e88cba17/forge-gui-desktop/src/main/java/forge/view/Main.java#L57

Reply:
Thank you, and for the run configuration without command-line arguments displayed below, I get the following crash report. Why do I need to specify decks when launching the GUI?

![image](https://github.com/Card-Forge/forge/assets/46533664/98b1ef7f-b65c-46e9-896f-313639ca37ca)

![image](https://github.com/Card-Forge/forge/assets/46533664/7875c3ea-5d26-487a-bbd4-976e84610ad9)

Reply:
you need add this to the Params, taken from the forge.sh:
https://github.com/Card-Forge/forge/blob/master/forge-gui-desktop/src/main/config/forge.sh#L46C10-L46C1026

```
--add-opens java.desktop/java.beans=ALL-UNNAMED --add-opens java.desktop/java.awt.color=ALL-UNNAMED --add-opens java.desktop/javax.swing.border=ALL-UNNAMED --add-opens java.desktop/javax.swing.event=ALL-UNNAMED --add-opens java.desktop/sun.awt.image=ALL-UNNAMED --add-opens java.desktop/sun.swing=ALL-UNNAMED --add-opens java.desktop/javax.swing=ALL-UNNAMED --add-opens java.desktop/java.awt=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.reflect=ALL-UNNAMED --add-opens java.base/java.text=ALL-UNNAMED --add-opens java.desktop/java.awt.font=ALL-UNNAMED --add-opens java.desktop/java.awt.image=ALL-UNNAMED --add-opens java.base/jdk.internal.misc=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/java.math=ALL-UNNAMED --add-opens java.base/java.util.concurrent=ALL-UNNAMED --add-opens java.base/java.net=ALL-UNNAMED -Dio.netty.tryReflectionSetAccessible=true 
```

Reply:
@Hanmac, when I supply the CLI arguments you provided, I get

Unknown mode.
Known mode is 'sim', 'parse'

Reply:
they need to be added to the VM Options

Reply:
@Hanmac , that did it, thank you! I guess the main thing left on this pull request is to work on finalize with @tool4ever .